drivers/scheduler/k8s/k8s.go

```diff
@@ -4328,8 +4328,40 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 				// check if the pvc has our VM as the owner
 				want := false
 				for _, ownerRef := range pvc.OwnerReferences {
-					if ownerRef.Kind == vm.Kind && ownerRef.Name == vm.Name {
-						want = true
+					if ownerRef.Kind == vm.Kind {
+						if ownerRef.Name == vm.Name {
+							want = true
+						}
+					}
+					// If volume is still not found, try the datavolumes
+					if !want && ownerRef.Kind == "DataVolume" {
+						// get datavolume
+						kvClient := k8sKubevirt.GetKubevirtClient()
+						dvList, err := kvClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
We should get only one datavolume that is referenced in the ownerRef. Otherwise, we may include volumes of another VM in the same namespace (in case there are multiple VMs in the same namespace).

Reply:
I am not sure I understand, even though we are getting all datavolumes in the VM namespace, in line# 4346 we are checking the datavolume owner reference against the VM name. That way we are ensuring we get only the datavolume that corresponds to the VM in question.

Reply:
Ah, yes. So, there is no correctness issue. Do you still want to just get a single data volume instead of listing multiple? We already know the name of the datavolume from the ownerRef. It is odd to use the "list" API when we can use "get".

Reply:
Fixed.